e3db/client.py

```diff
@@ -482,51 +482,100 @@ def register(self, registration_token, client_name, public_key, private_key=None
         """
-
-        url = "{0}/{1}/{2}/{3}/{4}/{5}".format(api_url, 'v1', 'account', 'e3db', 'clients', 'register')
-        if Crypto.get_mode() == 'nist':
-            payload = {
-                'token': registration_token,
-                'client': {
-                    'name': client_name,
-                    'public_key': {
-                        'p384': public_key
+        if public_signing_key == "":
+            url = "{0}/{1}/{2}/{3}/{4}/{5}".format(api_url, 'v1', 'account', 'e3db', 'clients', 'register')
+            if Crypto.get_mode() == 'nist':
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'p384': public_key
+                        }
+                    }
+                }
+            else:
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'curve25519': public_key
+                        }
                     }
                 }
-            }
         else:
-            payload = {
-                'token': registration_token,
-                'client': {
-                    'name': client_name,
-                    'public_key': {
-                        'curve25519': public_key
+            url = "{0}/{1}/{2}/{3}/{4}/{5}".format(api_url, 'v1', 'account', 'e3db', 'clients', 'register')
+            if Crypto.get_mode() == 'nist':
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'p384': public_key
+                        },
+                        'signing_key': {
+                            'ed25519': public_signing_key
+                        }
+                    }
+                }
+            else:
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'curve25519': public_key
+                        },
+                        'signing_key': {
+                            'ed25519': public_signing_key
+                        }                    
                     }
                 }
-            }
         response = requests.post(url=url, json=payload)
         self.__response_check(response)
         client_info = response.json()
         backup_client_id = response.headers['x-backup-client']
-
         if backup:
             if private_key is None:
                 raise RuntimeError("Cannot back up client credentials without a private key!")
 
```

Comment:
There needs to also be a check for the private signing key for v2 client backups